### ISSUE TITLE
Style colors aren't being applied to NavigationMapView as expected

### ISSUE THREAD

**Issue by johnnewman** [mapbox/mapbox-navigation-ios#4433]
### Mapbox Navigation SDK version

2.11.0

### Steps to reproduce

Hello, I'm using subclasses of DayStyle and NightStyle that customize some colors of the NavigationMapView appearance. After supplying the styles to a NavigationViewController via its `navigationOptions` initializer parameter, I'm seeing the appearance of the NavigationMapView class update, but not the corresponding color property of the `navigationMapView` instance.

Here is the Swift code to reproduce:
```Swift
class DayRedTraversedLineColor: DayStyle {
    override func apply() {
        super.apply()
        NavigationMapView.appearance().traversedRouteColor = .red
    }
}

class NightRedTraversedLineColor: NightStyle {
    override func apply() {
        super.apply()
        NavigationMapView.appearance().traversedRouteColor = .red
    }
}

...

class MyNavigationViewController {
    override func viewDidLoad() {
        super.viewDidLoad()
        let navigationController = NavigationViewController(
            for: IndexedRouteResponse(
                routeResponse: routeResponse,
                routeIndex: routeIndex
            ),
            navigationOptions: NavigationOptions(
                styles: [DayRedTraversedLineColor(), NightRedTraversedLineColor()]
            )
        )
        navigationController.view.translatesAutoresizingMaskIntoConstraints = false        
        navigationController.willMove(toParent: self)
        addChild(navigationController)
        view.addSubview(navigationController.view)
        view.constrain(subview: navigationController.view)
        navigationController.didMove(toParent: self)
        
        navigationController.routeLineTracksTraversal = true
        print("Appearance color: \(NavigationMapView.appearance().traversedRouteColor)")
        print("Actual color:     \(navigationController.navigationMapView?.traversedRouteColor)")
    }
}
```
And the output:
```
Appearance color: UIExtendedSRGBColorSpace 1 0 0 1
Actual color:     Optional(UIExtendedSRGBColorSpace 1 1 1 0)
```

The docs of `routeLineTracksTraversal` state:
> to customize the color that appears on the traversed section of a route, override the `traversedRouteColor` property for the `NavigationMapView.appearance()`.

I believe the color not matching what was set in the Style subclass is a bug.

Thank you!

### Expected behavior

The `traversedRouteColor` of the underlying `navigationMapView` matches the appearance set in the Style subclass provided to the NavigationViewController.

### Actual behavior

The color is using the default value, not the custom one.

### Is this a one-time issue or a repeatable issue?

repeatable

**Comment by kried** [comment]
Hi @johnnewman 

You could update your custom style code like this to apply `traversedRouteColor`.
```swift
let traitCollection = UIScreen.main.traitCollection
NavigationMapView.appearance(for: traitCollection).traversedRouteColor = .red
```
 You can check that the new value was applied with the changed logging code
```swift
print("Appearance color: \(NavigationMapView.appearance(for: .current).traversedRouteColor)")
print("Actual color:     \(navigationViewController.navigationMapView?.traversedRouteColor)")
```
Then you will be able to see the output
```
Appearance color: UIExtendedSRGBColorSpace 1 0 0 1
Actual color:     Optional(UIExtendedSRGBColorSpace 1 0 0 1)
```

Please take a look at [the docs about styling](https://docs.mapbox.com/ios/navigation/api/2.12.0/styling-the-ui.html) and [the sample](https://github.com/mapbox/mapbox-navigation-ios-examples/blob/d1f9c996c729944cfbcf96f124835f7f1eb3ac3c/Navigation-Examples/Examples/Styled-UI-Elements.swift#L71) for more information. Hope this helps.

**Comment by johnnewman** [comment]
Hi @kried, thank you for the insight! After updating my custom Style subclasses to use trait collections instead of `appearance`, I was able to see the proper color at runtime. Checking the color after the map is initialized within `viewDidLoad` didn't seem to print the proper RGBA values, but I got the expected RGBA values at runtime using the view debugger and printing out the colors there 👍. Going to close this ticket out.

I'm going to close this ticket down.
